samtranslator/model/sam_resources.py

```diff
@@ -348,7 +345,9 @@ def _construct_event_invoke_config(self, function_name, alias_name, lambda_alias
 
         return resources, policy_document
 
-    def _validate_and_inject_resource(self, dest_config, event, logical_id, conditions):  # type: ignore[no-untyped-def]
```

Comment:
note: this refactor this function to make it actually return destination not a wrapped `{"Destination": destination}`